tests/suites/test_suite_gcm.function

```diff
@@ -123,11 +124,16 @@ void gcm_encrypt_and_tag( int cipher_id, data_t * key_str,
         for( n1 = 0; n1 <= src_str->len; n1 += 1 )
         {
             mbedtls_test_set_step( n1 );
-            if( !check_multipart( &ctx, MBEDTLS_GCM_ENCRYPT,
-                                  iv_str, add_str, src_str,
-                                  dst, tag,
-                                  n1 ) )
-                goto exit;
+
+            for( n1_add = 0; n1_add <= add_str->len; n1_add += 1 )
+            {
+                mbedtls_test_set_step( n1_add );
```

Comment:
The point of the test step is to convey the conditions in which a test failure happens. When tests have no loop or auxiliary functions, the test case name and the line number are enough. The step number provides additional information when the same code line runs more than once for a test case. With a simple loop, the iteration number is enough information. With a nested loop, the step number should encode the step number from both loops. Here, with a test failure, we'd only see the problematic value of `n1_add` and not the problematic value of `n1`.

The step number is printed out in decimal. Following a precedent in `tests/suites/test_suite_ssl.function`:
```
mbedtls_test_set_step( n1 * 10000 + n1_add );
```

Same in the decrypt function.